### DISCUSSION TITLE
What's the difference between an ESP chip with Tasmota vs a Raspberry Pi Pico(2)?

### DISCUSSION THREAD

**Discussion by wayner9** [arendst/Tasmota#21974] [Q&A]
Maybe this is a dumb question, but are these similar devices?  If so what is the advantage of using an ESP device vs an RPi Pico?

**Comment by joba-1** [comment]
Not dumb, just very basic.
ESPs and the Pico-W are quite similar (MHz, RAM, flash, WLAN and bluetooth, arduino compatible, price)
ESP has way more support because it is available for much longer. Many libs are written for it or used by it for a long time while on the pico they should work but maybe no one has ever tried. OTOH I have written a bluetooth receiver/dac audio firmware for the Pico-W which works rock solid but I was never able to produce a stable (over long time) bluetooth app on an ESP - might be my fault, who knows... Also jitter free timers, e.g. for creating exact wave forms is not the strength of ESP's while the pico has a separate unit for that. So I prefer pico-w for audio, bluetooth and things requiring exact timing (microseconds) and ESP's for the rest.
Regarding tasmota the basic difference ist easy: ESP only.

**Comment by s-hadinger** [comment]
I would add that the main difference is that espressif chips have add integrated wifi from the beginning. Hence the device ecosystem is large enough that Tasmota can benefit from a large variety of commercial and semi-commercial devices

**Reply by wayner9** [reply]
Makes sense.  I wonder if smart device makers will start to use the Pi in the future?  IMO there is a bit of a barrier to installing Tasmota on a lot of devices as you have to take them apart and often have to make tricky connections to pins.  It would be nice if all of these devices had accessible USB ports and you could change the firmware to something like Tasmota with a laptop and a USB cable.

**Reply by s-hadinger** [reply]
It depends on manufacturers. Some like Athom are maker friendly. Others makes it more difficult. 
The bad side effect of Matter, is that any Matter certified device has its firmware locked. If they are esp32 based, you just can't flash them with anything else, because I of firmware encryption

**Reply by sfromis** [reply]
I'd not expect device makers to move towards the Raspberry Pi Pico 2 (RP2350), the chip is just too big and expensive for IoT, especially as it has no built-in radios. The latter is a major drawback.

The Pico may be great for hobbyists tinkering without needing radios, and Raspberry has a decent ecosystem for projects there. Way different from mass market consumer-oriented devices.

With current ESP32-Sx and Cx devices, it is very easy for device vendors to provide an USB interface if they so choose, as this is built into the chips, as long as it is not the original ESP32 or even ESP8266.
